keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -73,255 +73,208 @@ private void verifyEnvironmentVariables() {
+    assertThat(secretNames)
+        .contains(secretName1, secretName2)
+        .doesNotContain(secretName3, secretName4);
   }
 
+  // secretsWithMatchingValuesAreReturned: search([], [vB, vC]) returns [secret2, secret3, secret4]
   @Test
-  void listAndMapMultipleSecretsWithMultipleTags() {
-    createSecret(true, true, false);
-
+  void listAndMapSecretsWithMatchingTagValues() {
     final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
-            AbstractMap.SimpleEntry::new);
-
-    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+            Collections.emptyList(), List.of("tagValB", "tagValC"), AbstractMap.SimpleEntry::new);
+
+    final List<String> secretNames =
+        secretEntries.stream()
+            .map(entry -> String.valueOf(entry.getKey()))
+            .collect(Collectors.toList());
+    assertThat(secretNames)
+        .contains(secretName2, secretName3, secretName4)
+        .doesNotContain(secretName1);
   }
 
+  // secretsWithMatchingKeysOrValuesAreReturned: search([k1], [vB]) returns [secret2]
   @Test
-  void listAndMapMultipleSecretsWithSharedTags() {
-    createSecret(true, false, true);
+  void listAndMapSecretsWithMatchingTagKeysAndValues() {
 
     final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
-            AbstractMap.SimpleEntry::new);
-
-    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+            List.of("tagKey1"), List.of("tagValB"), AbstractMap.SimpleEntry::new);
+
+    final List<String> secretNames =
+        secretEntries.stream()
+            .map(entry -> String.valueOf(entry.getKey()))
+            .collect(Collectors.toList());
+    assertThat(secretNames)
+        .contains(secretName2)
+        .doesNotContain(secretName1, secretName3, secretName4);
   }
 
   @Test
-  void listAndMapMultipleSecretsWithMultipleAndSharedTags() {
-    createSecret(true, false, true);
-    createSecret(true, true, false);
-
-    final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
-        awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
-            AbstractMap.SimpleEntry::new);
-
-    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
-  }
-
-  @Test
-  void throwsAwayObjectsThatFailMapper() {
-    createSecret(true, false, false);
-
-    final String failEntryName = secretNames.get(1);
-
+  void throwsAwayObjectsWhichMapToNull() {
```

Comment:
What happens if the mapper throws an exception? Can we have a test to show that the AwsSecretsManager handles that

Reply:
Wasn't as clear about this as I should have been. I meant if the user supplied mapper function to mapSecrets fails for any reason we should handle that and continue process other secrets with the mapping function.